### PR TITLE
REF/TST: Add more pytest idiom to indexing/multiindex/test_getitem.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
xref #24040

for this pass:

- the Test classes have been removed.

- `catch_warnings`  used for the `.ix` deprecation replaced with `@pytest.mark.filterwarnings("ignore:\\n.ix:DeprecationWarning")` at test level.

cc @jreback @gfyoung

### CODE REVIEW

File: pandas/tests/indexing/multiindex/test_getitem.py [hunk 1]
```diff
@@ -11,335 +9,337 @@
...
+                                  ('b', 'two')])
+    s = Series([1, 2, 3, 4], index=idx)
+    s.index.set_names(['L1', 'L2'], inplace=True)
+    result = s.xs('one', level='L2')
+    expected = Series([1, 3], index=['a', 'b'])
+    expected.index.set_names(['L1'], inplace=True)
+    tm.assert_series_equal(result, expected)
+
+
+def test_getitem_duplicates_multiindex():
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_getitem.py]: can you try to parameteize this on the input frame / expected output.

Comment (jreback) [reply pandas/tests/indexing/multiindex/test_getitem.py]: might make sense to split out the empty case to a new test

File: pandas/tests/indexing/multiindex/test_getitem.py [hunk 2]
```diff
@@ -11,335 +9,337 @@
...
+
+    # GH5684
+    idx = MultiIndex.from_tuples([('a', 'one'), ('a', 'two'), ('b', 'one'),
+                                  ('b', 'two')])
+    s = Series([1, 2, 3, 4], index=idx)
+    s.index.set_names(['L1', 'L2'], inplace=True)
+    result = s.xs('one', level='L2')
+    expected = Series([1, 3], index=['a', 'b'])
+    expected.index.set_names(['L1'], inplace=True)
+    tm.assert_series_equal(result, expected)
```
Comment (gfyoung) [inline pandas/tests/indexing/multiindex/test_getitem.py]: Can we break this massive test up into smaller ones?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/multiindex/test_getitem.py [hunk 3]
```diff
@@ -11,335 +9,337 @@
...
+    # but not if it's mixed-type
+    df['foo', 'four'] = 'foo'
+    df = df.sort_index(level=0, axis=1)
+
+    # this will work, but will raise/warn as its chained assignment
+    def f():
+        df['foo']['one'] = 2
+        return df
+
+    pytest.raises(com.SettingWithCopyError, f)
```
Comment (gfyoung) [inline pandas/tests/indexing/multiindex/test_getitem.py]: Any error message worth checking?

File: pandas/tests/indexing/multiindex/test_getitem.py [hunk 4]
```diff
@@ -11,335 +9,337 @@
...
+    result = s[:, 0]
+    expected = Series([1], index=[0])
+    tm.assert_series_equal(result, expected)
+
+    result = s.loc[:, 1]
+    expected = Series([2, 3], index=[1, 2])
+    tm.assert_series_equal(result, expected)
+
+    # xs
+    result = s.xs(0, level=0)
```
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/tests/indexing/multiindex/test_getitem.py]: we could then paramatrize..

```
@pytest.mark.parametrize('access_method', [lambda s, x: s[:, x],
                                           lambda s, x: s.loc[:, x],
                                           lambda s, x: s.xs(x, level=1)])
@pytest.mark.parametrize('level1_value, expected', [
    (0, Series([1], index=[0])),
    (1, Series([2, 3], index=[1, 2]))
])
def test_series_getitem_multiindex(access_method, level1_value, expected):

    # GH 6018
    # series regression getitem with a multi-index

    s = Series([1, 2, 3])
    s.index = MultiIndex.from_tuples([(0, 0), (1, 1), (2, 1)])
    result = access_method(s, level1_value)
    tm.assert_series_equal(result, expected)
```

or is this OTT.

[the remark this answers is not in the document]
Comment (jreback) [reply pandas/tests/indexing/multiindex/test_getitem.py]: no, you certainly parameterize this

**Comment by simonjayhawkins** [comment]
@jreback @gfyoung 

`test_series_getitem_multiindex` and `test_getitem_duplicates_multiindex` split and parametrized. error messages checked.

carry on and split and parametrize the remaining tests in this file within this PR or limit the diff?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @simonjayhawkins keep em coming! I don't mind bigger ones, main issue is there are some other parts moving so easiest to get them in in self-contained units.

**Pull request merged**